torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -505,9 +505,63 @@ Tensor masked_fill_backward(const Tensor& grad, const Tensor& mask) {
       : grad.masked_select(mask).sum();
 }
 
-Tensor mul_tensor_backward(Tensor grad, Tensor other, ScalarType self_st) {
-  auto out = grad * other.conj();
-  return handle_r_to_c(self_st, out);
+Tensor mul_tensor_backward(
+    const Tensor& grad,
+    const Tensor& other,
+    bool self_is_sparse,
+    IntArrayRef self_sizes,
+    ScalarType self_st) {
+  // If a.is_sparse and b.is_sparse, then mul(a, b) expects:
+  // 1. a.dim() == b.dim(),
+  // 2. a.shape[:a.sparse_dim] == b.shape[:b.sparse_dim].
+  // Note, however, that mul(a, b) will handle broadcasting in dense dims.
+  // Autograd, however, will have issues with that and will try running
+  // either a.sum(d=...) or b.sum(d=...) to propagate sparse grads of the
+  // same shape as the inputs'. However, sum(d=...) is not implemented
```

Comment:
> sum(d=...) is not implemented for sparse tensors.

We do have [torch.sparse.sum(..., d=...)](https://pytorch.org/docs/master/generated/torch.sparse.sum.html#torch.sparse.sum) support. Can you be more specific, what is not implemented in `sum(d=...)` for COO tensors?

Reply:
This is not what `autograd` is using under the hood.

Reply:
>  I wonder why do we have `torch.sparse.sum` and not `Tensor.sum`?

Semantically, these are the same. Because sparse support is beta, the short-term plan has been to keep the strided and sparse tools separate.

I wonder if merging `torch.sparse.sum` to `Tensor.sum` would make supporting autograd for sparse tensors (including CSR, etc) easier? If so, that would be a good reason to advocate the merge.

Reply:
This is what `autograd` is running to reduce `grad` to the shape of the variable this `grad` is coming to. So, yes, that would make a lot of sense.

Reply:
@pearu , unfortunately, this does not work, as `sparse.sum` does not have `keepdim`. Alternatively, I could use `_sparse_sum` and then unsqueeze, but would it make the code look any better?

Reply:
> but would it make the code look any better?

This is not the goal here.

Reply:
The goal would be to understand what is needed for sparse autograd support. If this investigation is non-trivial, consider it OT within this PR.

Reply:
@pearu , `torch._masked.sum` is a high-level function implemented in Python and it calls to `values.sum` anyway.